cmd/otelcol/main.go

```diff
@@ -41,9 +47,17 @@ func main() {
 		GitHash:  version.GitHash,
 	}
 
-	svc, err := service.New(service.Parameters{ApplicationStartInfo: info, Factories: factories})
-	handleErr("Failed to construct the application", err)
+	return service.Parameters{Factories: factories, ApplicationStartInfo: info}, nil
+}
 
-	err = svc.Start()
-	handleErr("Application run finished with error", err)
+func runInteractive(params service.Parameters) {
+	app, err := service.New(params)
+	if err != nil {
+		log.Fatalf("failed to construct the application: %v", err)
+	}
+
+	err = app.Start()
```

Comment:
rant: we should really rename this function to `Run` (change #615 was rejected, but there were quite a few other breaking changes in the codebase since then).

Reply:
Yea this is admittedly particularly confusing in this file now since we wait for the response of `app.Start` in the `windowsservice.stop` function. But if we rename this, I think it should be done in a separate PR